mlir/lib/Dialect/Rock/Transforms/ConvToGemm.cpp

```diff
@@ -267,41 +269,58 @@ static Value makeGpuAllocContaining(OpBuilder &b, Value v) {
 
 /// 0-initialize a given buffer.
 struct ZeroInitKernelRewritePattern final
-    : public OpConversionPattern<ZeroInitKernelOp> {
-  using OpConversionPattern<ZeroInitKernelOp>::OpConversionPattern;
+    : public OpConversionPattern<InitKernelOp> {
+  using OpConversionPattern<InitKernelOp>::OpConversionPattern;
 
-  LogicalResult matchAndRewrite(ZeroInitKernelOp op,
-                                ZeroInitKernelOpAdaptor adaptor,
+  LogicalResult matchAndRewrite(InitKernelOp op, InitKernelOpAdaptor adaptor,
                                 ConversionPatternRewriter &b) const override {
     Location loc = op.getLoc();
     TypedValue<ShapedType> buffer = op.getBuffer();
     Type bufferType = buffer.getType().getElementType();
     if (!op.getElemsPerThread().has_value())
       return op->emitOpError("elems per thread not set");
 
-    int64_t zeroInitVectorLen = getUtilityVectorizationLen(
+    int64_t initVectorLen = getUtilityVectorizationLen(
         buffer.getType(), op.getElemsPerThread()->getZExtValue());
-    Type storeType = vectorTypeOrSelf(bufferType, zeroInitVectorLen);
+    Type storeType = vectorTypeOrSelf(bufferType, initVectorLen);
     bool needs64BitIdx = is4GBMemoryType(buffer.getType());
-    Value zeroOp = createZeroConstantOp(b, loc, storeType);
+
+    Type elementType = mlir::getElementTypeOrSelf(storeType);
+    Value initOp;
+    auto initValueAttr = op.getInitValueAttr();
+    if (initValueAttr) {
+      if (auto floatInitValueAttr = initValueAttr.value().cast<FloatAttr>()) {
+        auto initValue = floatInitValueAttr.getValue().convertToFloat();
+        initOp =
```

Comment:
Check if this works with vectors - I don't remember if it does

Reply:
Hi @krzysz00, I tried `vector::SplatOp`. I noticed that the operation requires `type` and `arith::ConstantOp`. I tried to pass `TypedAttr` or `APFloat/APInt` to the constructor, but it didn't work

Reply:
Ah. Well then we'll clean this up Later (tm), it's not a big deal